src/Components/Components/src/OwningComponent.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// A base class that creates a service provider scope, and resolves a service of type <typeparamref name="TService"/>.
+    /// </summary>
+    /// <typeparam name="TService">The service type.</typeparam>
+    /// <remarks>
+    /// Use the <see cref="OwningComponent{TService}"/> class as a base class to author components that control
+    /// the lifetime of a service or multiple services. This is useful when using a transient or scoped service that
```

Comment:
"multiple services"? Does this address that scenario? There's only one generic type param.

Reply:
Oh I see, you use `ScopedServices`.

If you require two services, and you consider them equally important, it would be strange to pick one of them to be `TService` and the other not. Should we also have a non-generic variant of `OwningComponent` that just gives you the `ScopedServices`? Then `OwningComponent<TService>` would be a very thin subclass of that.

Reply:
Yeah I suppose we could do that. I originally avoided it because you can't tell the difference between generic and non-generic when used in markup, but since this is a base class it will never appear in markup.